TrafficCapture/dockerSolution/src/main/docker/migrationConsole/lib/console_link/tests/test_cluster.py

```diff
@@ -71,6 +69,53 @@ def test_missing_endpoint_refused():
     assert excinfo.value.args[1]['cluster'][0]["endpoint"] == ["required field"]
 
 
+def test_valid_cluster_with_secrets_auth_created():
+    valid_with_secrets = {
+        "endpoint": "https://opensearchtarget:9200",
+        "allow_insecure": True,
+        "basic_auth": {
+            "username": "admin",
+            "password_from_secret_arn": "arn:aws:secretsmanager:us-east-1:12345678912:secret:master-user-os-pass"
+        },
+    }
+    cluster = Cluster(valid_with_secrets)
+    assert isinstance(cluster, Cluster)
+
+
+def test_invalid_cluster_with_two_passwords_refused():
+    two_passwords = {
+        "endpoint": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
+        "allow_insecure": True,
+        "basic_auth": {
+            "username": "XXXXX",
+            "password": "XXXXXXXXXXXXXX",
+            "password_from_secret_arn": "arn:aws:secretsmanager:us-east-1:12345678912:secret:master-user-os-pass"
+        },
+    }
+    with pytest.raises(ValueError) as excinfo:
+        Cluster(two_passwords)
+    assert "Invalid config file for cluster" in excinfo.value.args[0]
+    assert excinfo.value.args[1]["cluster"][0]['basic_auth'] == [
+        "More than one value is present: ['password', 'password_from_secret_arn']"
```

Comment:
I'm assuming that this specific messaging is generated by `cerberus`.  I'm concerned this places us at risk of having our unit tests fail if they decide to slightly change their formatting, as we're tightly coupled to an implementation detail we don't control.

I'm curious what value-add you see in checking for the specific string on top of a `ValueError` being thrown?  It feels like we should either trust our library to do the right thing or, if we don't trust it (or we want to make sure it's failing for the reasons we expect) we should be catching the exception thrown by `cerberus` at a lower level and re-throwing an exception we control.

I imagine we do this in all of our unit tests, so changing it would be a substantial task.

Reply:
I see what you mean. We show these messages to users, so I care about more than just ValueError being thrown, I wanted to make sure that there was enough information to be actionable. So I started by testing for the pieces of the message I care about (is 'password' present, is 'password_from_secret_arn' present?), and ended up simplifying it to just match the whole message.

But you're right that it is more fragile. I'd rather have our tests break than suddenly start showing users a less helpful message without our awareness.

And if we're catching the exception thrown by cerberus, validating that it matches a specific string ("more than one value is present", for instance) and then rethrowing a custom exception, won't all the same tests break if they change their wording? I agree that it's better to have to fix it only in one spot, but I'm not sure it's _that_ much better.

Reply:
Another thing that occurred to me - it probably isn't the best user experience to present this info to the user in this way.  Are we just killing the process and dumping the exception to stdout?  If I'm a user and I see a stack trace from the tool I just ran, I'd normally assume there was a bug or something broken inside the tool.  Typically, I'd expect to see log messages saying I did something wrong, not a stack trace.

It feels like our overall experience here isn't where we want it to be longer term, and that's fine for this PR, but we should be moving a different direction going forward and that's worth talking about.

Reply:
If you have specific thoughts for improving it, happy to put them into a JIRA.

Reply:
Per discussion w/ @mikaylathompson - it seems like there's a way to hook into `cerberus` at a lower level and present our own messaging for situations such as this.  This would strike a better balance between leveraging the library and retaining control over the user experience by removing the chance that an update to `cerberus` would break our tests and, potentially, user scripting relying on specific phrases provided by the library.

@mikaylathompson will make a followup task to investigate that approach in the future.